pyptex/__init__.py

```diff
@@ -229,298 +241,305 @@ def genname(self,pattern="fig{gencount}.eps"):
         `pyp.genname()` does not actually create the file. `pyp.genname()` increments
         `pyp.gencount` every time it is called.
         """
-        self.gencount = self.gencount+1
-        return self.gendir+'/'+pattern.format(**self.__dict__)
+        self.gencount += 1
+        return f'{self.gendir}/{pattern.format(**self.__dict__)}'
 
-    def savefig(self,fig,pattern="fig{gencount}.eps",**kwargs):
+    def savefig(self, fig, pattern: str = 'fig{gencount}.eps', **kwargs):
         """Save a figure to the a-generated/* subdirectory.
-        
+
         If `pyp` is an object of type `pyptex`:
-        `pyp.savefig(fig)` saves a SymPy or matplotlib figure to the `a-generated/*` 
-        subdirectory, using the `pyptex.genname()` automatically generated filename. 
-        `pyp.savefig(fig,pattern)` further specifies the filename pattern of the
+        `pyp.savefig(fig)` saves a SymPy or matplotlib figure to the `a-generated/*`
+        subdirectory, using the `pyptex.genname()` automatically generated filename.
+        `pyp.savefig(fig, pattern)` further specifies the filename pattern of the
         generated name, see `genname()`.
-        `pyp.savefig(fig,pattern,...)` passes any further keyword arguments directly
+        `pyp.savefig(fig, pattern, ...)` passes any further keyword arguments directly
         to the `savefig` function from matplotlib.
 
         A typical way of using this from a TeX file is:
         `\\includegraphics{@{pyp.savefig(...)}}`
         """
-        if(self.__sympy_plot__ == None):
-            self.__sympy_plot__ = sympy.plotting.plot(1,show=False).__class__
+        if self.__sympy_plot__ is None:
+            self.__sympy_plot__ = sympy.plotting.plot(1, show=False).__class__
         figname = self.genname(pattern)
-        if(fig.__class__==self.__sympy_plot__):
+        if fig.__class__ == self.__sympy_plot__:
             backend = fig.backend(fig)
             backend.process_series()
             backend.fig.savefig(figname, **kwargs)
         else:
             fig.savefig(figname, **kwargs)
         self.dep(figname)
         return figname
+
     def generateddir(self):
-        """This is an internal function that creates the generated directory"""
-        self.gendir = self.filename+'-generated'
+        """This is an internal function that creates the generated directory."""
+        self.gendir = f'{os.fspath(self.filename)}-generated'
```

Comment:
I'm not sure I understand the purpose of this os.fspath here, I've never used this function before, but it seems to be mostly a way to ensure that self.filename is a string to begin with? Why did you use os.fspath here and not elsewhere? If it's not for any specific reason, out of a desire for simplicity, I would just leave it out
```suggestion
        self.gendir = f'{self.filename}-generated'
```

Reply:
Oops! I was going to try converting everything path-like to strings, but stopped because I wasn't sure that I could cover everything and didn't want to break things. This is a remnant of that that I forgot to remove.